packages/relay/src/lib/services/hapiService/hapiService.ts

```diff
@@ -0,0 +1,361 @@
+
+    return this.client;
+  }
+
+  /**
+   * Return true if reinitialization feature is enabled.
+   * @returns isEnabled boolean
+   */
+  public getIsReinitEnabled() {
+    return this.isReinitEnabled;
+  }
+
+  /**
+   * Return transaction count with current sdk instance.
+   * @returns transactionCount
+   */
+  public getTransactionCount() {
+    return this.transactionCount;
+  }
+
+  /**
+   * Return error codes which can trigger a sdk instance reinitialization.
+   * @returns errorCodes
+   */
+  public getErrorCodes() {
+    return this.errorCodes;
+  }
+
+  /**
+   * Return time until reset of the current sdk instance.
+   */
+  public getTimeUntilReset() {
+    return this.resetDuration - Date.now();
+  }
+
+  /**
+   * Initialize consensus node cost metrics
+   * @param {Registry} register
+   * @returns {Histogram} Consensus node cost metric
+   */
+  private initCostMetric(register: Registry) {
+    const metricHistogramCost = 'rpc_relay_consensusnode_response';
+    register.removeSingleMetric(metricHistogramCost);
+    return new Histogram({
+        name: metricHistogramCost,
+        help: 'Relay consensusnode mode type status cost histogram',
+        labelNames: ['mode', 'type', 'status', 'caller', 'interactingEntity'],
+        registers: [register]
+    });
+  }
+
+  /**
+   * Initialize consensus node gas metrics
+   * @param {Registry} register
+   * @returns {Histogram} Consensus node gas metric
+   */
+  private initGasMetric(register: Registry) {
+    const metricHistogramGasFee = 'rpc_relay_consensusnode_gasfee';
+    register.removeSingleMetric(metricHistogramGasFee);
+    return new Histogram({
+        name: metricHistogramGasFee,
+        help: 'Relay consensusnode mode type status gas fee histogram',
+        labelNames: ['mode', 'type', 'status', 'caller', 'interactingEntity'],
+        registers: [register]
+    });
+  }
+
+  /**
+   * Initialize operator account metrics
+   * @param {Client} clientMain
+   * @param {Logger} logger
+   * @param {Registry} register
+   * @returns {Gauge} Operator Metric
+   */
+  private initOperatorMetric(clientMain: Client, logger: Logger, register: Registry) {
+    const metricGaugeName = 'rpc_relay_operator_balance';
+    register.removeSingleMetric(metricGaugeName);
+    return new Gauge({
+        name: metricGaugeName,
+        help: 'Relay operator balance gauge',
+        labelNames: ['mode', 'type', 'accountId'],
+        registers: [register],
+        async collect() {
+            // Invoked when the registry collects its metrics' values.
+            // Allows for updated account balance tracking
+            try {
+                const accountBalance = await (new AccountBalanceQuery()
+                    .setAccountId(clientMain.operatorAccountId!))
+                    .execute(clientMain);
+                this.labels({ 'accountId': clientMain.operatorAccountId!.toString() })
+                    .set(accountBalance.hbars.toTinybars().toNumber());
```

Comment:
Since we are refactoring this I would like  to  suggest to fetch accountBalance of the  operator from the mirror node? since getting  it from the the consensus node costs money?
Also,  the sdkClient does  not even use this  `Gauge` so no need  to  pass it along with the other metrics... not even sure, if could be moved to the `Relay.ts` itself so to not bring mirror node dependencies to `hapiService.ts` at all.

Reply:
Hmm, where is this gauge used? I'm not sure if `Relay.ts` also shouldn't know and take care about metrics, especially in the constructor. 
Agree that we should fetch balance from mirror node, if possible and not burden this service with mirror node details.
Since we have quite a lot of metrics now, I'm thinking about a metrics service, which takes care of all metric creations and assignments, what do you think?

Reply:
I've moved it over in the relay class, but only temporary, until I'm ready with the metrics service class.